File: practice/algorithms/greedy/max_min/max_min.py
#!/usr/bin/env python

n = int(raw_input().strip())
k = int(raw_input().strip())
buff = [int(raw_input().strip()) for _ in range(0, n)]
buff.sort()

res = buff[k - 1] - buff[0]
for POS in range(1, n - k + 1):
    diff = buff[POS + k - 1] - buff[POS]
    if diff < res:
        res = diff
print res

File: practice/algorithms/greedy/max_min/max_min.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <climits>

using namespace std;

int main()
{
    int n, k;
    cin >> n;
    cin >> k;
    
    vector<int> buff(n);
    for (int pos = 0; pos < n; ++pos)
    {
        cin >> buff[pos];
    }
    
    sort(buff.begin(), buff.end());
    
    int res = buff[k - 1] - buff[0];
    for (int pos = 1; pos < n - k + 1; ++pos)
    {
        int diff = buff[pos + k - 1] - buff[pos];
        if (diff < res)
        {
            res = diff;
        }
    }
    
    cout << res << endl;
    
    return 0;
}

